### PR TITLE
fix  TestEventChannelFull random fail

### LINKED ISSUE #76989: //pkg/kubelet/pleg:go_default_test is flaky
@kubernetes/sig-node-bugs 

```
FAIL: //pkg/kubelet/pleg:go_default_test (see /usr/local/google/home/fejta/.cache/bazel/_bazel_fejta/631ceaac9e4091781154c1030ef08d3e/execroot/__main__/bazel-out/k8-fastbuild/testlogs/pkg/kubelet/pleg/go_default_test/test_attempts/attempt_1.log)

FLAKY: //pkg/kubelet/pleg:go_default_test (Summary)
      /usr/local/google/home/fejta/.cache/bazel/_bazel_fejta/631ceaac9e4091781154c1030ef08d3e/execroot/__main__/bazel-out/k8-fastbuild/testlogs/pkg/kubelet/pleg/go_default_test/test.log
      /usr/local/google/home/fejta/.cache/bazel/_bazel_fejta/631ceaac9e4091781154c1030ef08d3e/execroot/__main__/bazel-out/k8-fastbuild/testlogs/pkg/kubelet/pleg/go_default_test/test_attempts/attempt_1.log
INFO: From Testing //pkg/kubelet/pleg:go_default_test:
==================== Test output for //pkg/kubelet/pleg:go_default_test:
E0424 06:42:20.674764      18 generic.go:274] event channel is full, discard this relist() cycle event
--- FAIL: TestEventChannelFull (0.00s)
    generic_test.go:97: Actual events differ from the expected; diff:
         [{"ID":"1234","Type":"ContainerRemoved","Data":"c1"},{"ID":"1234","Type":"ContainerDied","Data":"c2"},{"ID":"
        
        A: 1234","Type":"ContainerStarted","Data":"c3"},{"ID":"4567","Type":"ContainerRemoved","Data":"c1"}]
        
        B: 4567","Type":"ContainerRemoved","Data":"c1"},{"ID":"4567","Type":"ContainerStarted","Data":"c4"}]
        
FAIL
```

### PR DESCRIPTION
**Pull request by changyaowei**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>

> /kind failing-test

**What this PR does / why we need it**:
In unit test ` TestEventChannelFull`, it will random fail, because the discard event is random, so when we check event , we should use include but not equal.

**Which issue(s) this PR fixes**:

Fixes #76989

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Fix TestEventChannelFull random fail
```

### CODE REVIEW

**Comment by changyaowei** [comment]
@apelisse @resouer  PTAL

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

My one question is why we aren't seeing more consistent flakes because of this? Is it because, while Golang doesn't guarantee the ordering when iterating through a dictionary, it is "in practice" reproducible?

I think this change makes sense either way, I just want to be sure I'm understanding why we haven't been seeing more flakes.

**Comment by apelisse** [comment]
I know that bazel runs the tests multiple times to detect flakes, so they're not "showing up there" (they are shown as flakes, but still pass with success)

**Comment by changyaowei** [comment]
By test, we can find that Golang can't guarantee the ordering when iterating through a dictionary.

**Comment by mattjmcnaughton** [comment]
Ah gotcha - good stuff, still lgtm :)

**Comment by mattjmcnaughton** [comment]
ping @Random-Liu - I think fixing this issue will address https://github.com/kubernetes/kubernetes/issues/76989 :)

@changyaowei do you want to update the PR description accordingly to reference this fixes #76989

**Comment by changyaowei** [comment]
> ping @Random-Liu - I think fixing this issue will address #76989 :)
> 
> @changyaowei do you want to update the PR description accordingly to reference this fixes #76989

Already update, thanks for you advise

File: pkg/kubelet/pleg/generic_test.go [hunk 1]
```diff
@@ -217,9 +217,17 @@ func TestEventChannelFull(t *testing.T) {
 		{ID: "1234", Type: ContainerDied, Data: "c2"},
 		{ID: "1234", Type: ContainerStarted, Data: "c3"},
 		{ID: "4567", Type: ContainerRemoved, Data: "c1"},
+		{ID: "4567", Type: ContainerStarted, Data: "c4"},
 	}
 	actual = getEventsFromChannel(ch)
-	verifyEvents(t, expected, actual)
+	assert.True(t, len(actual) == 4, "channel length should be 4")
+	for _, actualEvent := range actual {
+		for _, expectedEvent := range expected {
```
Comment (Random-Liu) [inline pkg/kubelet/pleg/generic_test.go]: Can we use `assert.Contains`? I remember it will do deep equal when comparing.

Comment (changyaowei) [reply pkg/kubelet/pleg/generic_test.go]: It can't work with  assert.contains, i use assert.Subsetf instead

**Comment by mattjmcnaughton** [comment]
@Random-Liu @changyaowei it'd be great to merge this if possible, because this flake just got reported in #sig-node. I think we run the risk of multiple fixing this will this remains open.

Thanks to both of you for your work on this :)

**Pull request merged**